packages/Menu/src/Menu.tsx

```diff
@@ -0,0 +1,173 @@
+  box-shadow: 0 2px 5px rgba(0, 0, 0, 0.15);
+  background-color: ${colors.mongodb.white};
+`;
+
+interface Props {
+  /**
+   * Determines the active state of the Menu.
+   *
+   * default: `false`
+   */
+  active: boolean;
+
+  /**
+   * Determines the alignment of the Menu relative to a trigger element.
+   *
+   * default: `bottom`
+   */
+  align: Align;
+
+  /**
+   * Determines the justification of the Menu relative to a trigger element.
+   *
+   * default: `end`
+   */
+  justify: Justify;
+
+  /**
+   * Content that will appear inside of the Menu.
+   */
+  children?: React.ReactElement;
+
+  /**
+   * className applied to Menu.
+   */
+  className?: string;
+
+  /**
+   * A reference to the element against which the Menu will be positioned.
+   */
+  refEl?: React.RefObject<HTMLElement>;
+
+  /**
+   * A slot for the element used to trigger the Menu. Passing a trigger allows
+   * Menu to control opening and closing itself internally.
+   */
+  trigger?: React.ReactElement;
+
+  /**
+   * Specifies that the popover content will appear portaled to the end of the DOM,
+   * rather than in the React subset of the DOM tree.
+   *
+   * default: `true`
+   */
+  usePortal?: boolean;
+}
+
+/**
+ * # Menu
+ *
+ * Menu component
+ *
+ * ```
+<button>
+  <Menu active={true}>
+    <MenuItem>Hello World!</MenuItem>
+  </Menu>
+</button>
+ * ```
+ * @param props.children Content to appear inside of Menu.
+ * @param props.active Boolean to describe whether or not Menu is active.
+ * @param props.className Classname applied to Menu.
+ * @param props.align Alignment of Menu relative to another element: `top`, `bottom`, `left`, `right`.
+ * @param props.justify Justification of Menu relative to another element: `start`, `middle`, `end`.
+ * @param props.refEl Reference element that Menu should be positioned against.
+ * @param props.usePortal Boolean to describe if content should be portaled to end of DOM, or appear in DOM tree.
+ * @param props.trigger Trigger element to set active state of Menu, makes component controlled
+ *
+ */
+
+function Menu({
+  align = 'bottom',
+  justify = 'end',
+  usePortal = true,
+  active,
+  children,
+  className,
+  refEl,
+  trigger,
+  ...rest
+}: Props) {
+  const [isActive, setActiveState] = useState(false);
+
+  const syntheticToggleEventHandler: EventHandler<React.SyntheticEvent> = e => {
+    e.nativeEvent.stopImmediatePropagation();
+    setActiveState(!isActive);
+  };
+
+  const nativeToggleEventHandler = (e: Event) => {
+    e.stopImmediatePropagation();
+    setActiveState(!isActive);
+  };
+
+  const handleEscape = (e: KeyboardEvent) => {
+    e.keyCode === 27 && nativeToggleEventHandler(e);
+  };
+
+  const popoverContent = (
+    <Popover
+      key="popover"
+      active={trigger ? isActive : active}
+      align={align}
+      justify={justify}
+      refEl={refEl}
+      usePortal={usePortal}
+    >
+      <div {...rest} className={cx(rootMenuStyle, className)} role="menu">
```

Comment:
Thought it made more sense that splatting them on the popover, but do you think that I should move it up?

Reply:
I _think_ splatting on the element you currently are is right, but could be convinced otherwise. To note: If we move the splat to Popover, we need to make sure that any passed style prop is still applied to this element.

Reply:
I _think_ splatting on the element you currently are is right, but could be convinced otherwise. To note: If we move the splat to Popover, we need to make sure that any passed style prop is still applied to this element.

Reply:
I'm wondering what props we want to pass through? What's a use case?

Maybe it's something we can add in a later release once we answer that question?

Reply:
I think the only props we'd actually want to pass through to Popover are the ones that are explicitly supported by the component (minus spacing). Any other props should probably go on the base display element here.

The primary reasons we'd want to splat at all is for consistency with how we handle the rest of the components, as well as `data-*` and `aria-*` props that might possibly be needed. I _think_ the place that you'd want those data and aria props to be rendered is usually going to be here, but I could be convinced otherwise.

Reply:
Yeah I agree that it feels like here is what's correct but feels like we're guessing too. I think I'm ok to keep it as is for now, and I think it's a sensible default, I'd feel overall better if we had a concrete use case though.

Reply:
I've been thinking of it as: Spread where someone may need an arbitrary attribute hook, or a handler we didn't anticipate needing, or perhaps a title attribute (all of which I think makes this the appropriate place to spread the props). In the checkbox component, we spread on the checkbox input itself despite it not being actually displayed, for instance. Personally, I think we are being deliberate where we're spreading the props (hence this discussion happening in the first place).

End-to-end tests and unit tests using libraries like `react-testing-library` will need to pass `data-*` to an element to interact with it. There may be a case where an app wants to use `aria-controls=""` to indicate a relationship (though I suppose this example is not great). Perhaps something should happen in the app when anything is clicked within the menu? We didn't need to explicitly add the onClick prop to the component for that to be possible.

All-in-all, I'm not convinced that we need a specific use case as this pattern is for props we didn't anticipate needing to pass through.

Reply:
Yeah, I think making this testable is a valid use case. Looking at the tests this definitely looks like the appropriate place to put the splat.

Splats always make me nervous in general, as it's functionality that's hard to take away in the future if we need to. So anytime I see its usage I get nervous.

This looks good to me though.

Reply:
Understood – we've committed to a fair bit of flexibility in our props in the name of wider adoption across teams, which has the downside of having some API's that are difficult to change in the future. I hear you and appreciate the reluctance in this case, and would love for there to be a better way to accommodate the wide use-cases we have across products.

If you have a pattern that you think might be better going forward, I'd love to hear it.